GitUpKit/Views/GIQuickViewController.m

```diff
@@ -235,3 +251,212 @@ - (BOOL)diffFilesViewController:(GIDiffFilesViewController*)controller handleKey
 }
 
 @end
+
+#import "GICommitListViewController.h"
+@interface GIQuickViewControllerWithCommitsList () <GICommitListViewControllerDelegate>
+@property (strong, nonatomic, readonly) GICommitListViewController *leftController;
+@property (strong, nonatomic, readonly) GIQuickViewController *rightController;
+
+@property (strong, nonatomic, readwrite) NSLayoutConstraint *hiddenConstraint;
+@property (strong, nonatomic, readwrite) NSLayoutConstraint *revealedConstraint;
+@end
+
+@implementation GIQuickViewControllerWithCommitsList
+
+#pragma mark - Accessors
+- (GICommitListViewController *)leftController {
+  return self.childViewControllers.firstObject;
+}
+
+- (GIQuickViewController *)rightController {
+  return self.childViewControllers.lastObject;
+}
+
+#pragma mark - Check
+- (BOOL)isHistoryShown {
+  return self.leftController.results.count > 0;
+}
+
+#pragma mark - Actions
+- (void)toggleLeftView {
+  BOOL shouldReveal = self.isHistoryShown;
+  self.hiddenConstraint.active = !shouldReveal;
+  self.revealedConstraint.active = shouldReveal;
+  [self.view setNeedsLayout:YES];
+//  [self.view setNeedsDisplay:YES];
+  [NSAnimationContext runAnimationGroup:^(NSAnimationContext * _Nonnull context) {
+    context.duration = 0.25;
+    context.allowsImplicitAnimation = YES;
+    [self.view layout];
+//    [self.view displayIfNeeded];
+  } completionHandler:^{
+  }];
+}
+
+#pragma mark - Layout
+- (void)addConstraints {
+  if (@available(macOS 10.11, *)) {
```

Comment:
Why have both? Does the older API not work on newer systems? If the older API works on newer systems, then just use the older API.

Reply:
I hope we could raise version to 10.11 with September release :)